adr/003-mediation-fees.md

```diff
@@ -0,0 +1,146 @@
+# Mediation Fees calculation and message format
+
+* Status: proposed
+* Deciders: Augusto, Dominik, Karl, Paul
+* Date: 2019-05-07
+
+## Context and Problem Statement
+
+Mediating nodes demand fees for their mediation services. These fees should not only incentivize running a Raiden node, but also help balancing the network. This can be done by reducing the mediation fees for payments which leave channels in a more balanced state than before the payment (possibly even to a negative fee).
+
+## Decision Drivers
+
+* Balancing the network should be incentivized
+* Raiden nodes will act selfishly
+* Mediation fees should be flexible enough to accommodate arbitrary business models for parties operation Raiden nodes.
+
+## Considered Options
+
+1. A fixed fee + a proportional fee (flat + prop * amount)
+2. A fixed fee + a proportional fee + an imbalance fee based on a fixed imbalance measure
+3. A fixed fee + a proportional fee + submitting points of a function describing the imbalance fee
+4. Submitting points of a single function (dependent on the payment amount) which includes all fees
+
+## Decision Outcome
+
+Chosen option: **Option 3**, but with some important additions (see below).
+
+Option 1 would be sufficient for the short very term, but does not incentivize rebalancing. Option 2 does not work because no objective balance measure could be found that is valid across all business models / user interests. Option 4 would work but require sending a changed fee schedule after every transfer or mediation. Since this information is broadcast, that's a non-negligible amount of traffic.
+
+### Positive Consequences
+
+* Adaptable to different business models
+* Both in- and outgoing channels can be rebalanced
+* Fully exhausting a channel can be punished more strongly than just moving into the wrong direction
+* Fee schedules only have to be changed when the node's strategy changes or the on-chain capacity changes via `deposit` or `withdraw`
+
+### Negative Consequences
+
+* Comparatively complex
+
+## Detailed analysis
+
+One important detail is that fees are not only calculated for the channel over which the mediation payment is sent, but also for the channel over which the payment is received. Without this, a node could not incentivize receiving through a certain channel, which is just as important for rebalancing as sending through the right channel. So the total mediation fee for a node is
```

Comment:
Maybe the example we had at the whiteboard would help to understand why this is desirable.

Reply:
Added a section for the example.